docs/architecture.rst

```diff
@@ -33,6 +34,64 @@ ActionSet. Finally, the original ActionSet will be updated by the
 controller with status and other metadata generated by the action
 execution.
 
+Kopia Repository Server Workflow
+================================
+
+Introducing Kopia
+------------------
+
+Kopia is a powerful, cross-platform tool for managing encrypted backups
+in the cloud. It provides fast and secure backups using compression,
+data `de-duplication`, and client-side end-to-end encryption. It supports
+a variety of backup storage targets, including object stores, which allows
+users to choose the storage provider that best addresses/meets their
+requirements. In Kopia, these storage locations are called repositories.
+It is a lock-free system that allows concurrent multi-client operations,
+including garbage collection. To explore other features of Kopia,
+see its `documentation <https://kopia.io/docs/features/>`_
+
+Kopia Repository Server
+-----------------------
+
+A Kopia Repository Server allows Kopia clients to proxy access to the backend storage
+location through it. At any given time, a repository server can only connect to a single
+repository. Due to this limitation, a separate instance of the server will be used
+for each repository.
+
+In Kanister, the server comprises of Kubernetes pod and service. The pod runs
+the Kopia repository server process that will be used by Kopia clients to perform
+backup and restore operations. Kopia clients only need a username/password and
+service name to connect to the server, without needing/requiring to know the backend
+storage location. This approach enhances security since only authorized users can
+access the Kopia repository server. These authorized users must be added to the
+server before starting the server.
```

Comment:
```suggestion
In Kanister, the server comprises of Kubernetes pod and service. The pod runs
the Kopia repository server process that will be used by Kopia clients to perform
backup and restore operations. Kopia clients only need a username/password and
service name to connect to the server without requiring to know the backend
storage location. This approach enhances security since only authorized users can
access the Kopia repository server. These authorized users must be added to the
server before starting the server.
```